tensorflow/python/keras/layers/kernelized_test.py

```diff
@@ -392,6 +393,19 @@ def test_good_kernel_approximation_multiple_inputs(self, initializer, scale,
     exact_kernel_matrix = exact_kernel_fn(x, y)
     self._assert_all_close(approx_kernel_matrix, exact_kernel_matrix, atol=0.05)
 
+  def testTrainLayer(self):
+    """Ensure the layer may be trained as part of a model."""
```

Comment:
You could use layer_test() which will do a bunch of extra verifications for a layer, eg saving/loading etc.

See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/testing_utils.py#L87

Reply:
Thanks for your suggestion - this looks really cool and is exactly what I need to use here. Unfortunately I am having some problems using this function with the layer - I am seeing errors that suggest the layer has not been registered. As soon as I can get this cleared up I will update the PR. My apologies for the delay.

Reply:
@johnamcleod, Any update on this PR? Please. Thanks!

Reply:
I was not able to figure out how to use the `layer_test` function with this layer, so I will close the PR. I apologise for the delay.